### PR TITLE
Add PDF Endpoint to API

### PR DESCRIPTION
**Pull request by dantheman999301**
Had a go at this myself to familiarise myself with the codebase.

Was not sure whether to abstract some of the common checks out into functions e.g. the flags parsing.

Let me know what you think 👍

### CODE REVIEW

**Comment by joelgriffith** [comment]
Hey @dantheman999301, thanks so much for the PR! You beat me to my initial pass at offering a `/screenshot` API. Once that lands I'd love to incorporate your PR, but there's gonna be some conflicts (likely), and a better way to schedule work vs doing a `queue.push` call. Want to get your thoughts on:

- Merging this work and I can take care of "migrating" it over to the new model. There'd be some API changes in the `/pdf` API that'd be breaking, but would offer the same functionality.

- I can merge in my work, and we can update this PR to adjust for the differences.

In any case, thanks again for the contribution! Let me know what you're thinking and I can proceed from there!

**Comment by joelgriffith** [comment]
I just pushed up master with how functions are setup, there's essentially three parts:

- Create a `pdf` function in the `/functions` directory (there's a few others there.
- Establish a JOI schema in `/src/schemas`. Should be an object with a `url` property + and `options` property that matches the `page.pdf` options (as long as they're serializable.
- Mount the route, validate the schema, and use the `this.runFunction` method to handle all the queueing/timeouts and so forth!

**Comment by dantheman999301** [comment]
Looks great! I'll get on this today.

File: functions/pdf.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * pdf function
+ *
+ * Example invocation:
+ *
```
Comment (joelgriffith) [inline functions/pdf.js]: Nice detail here on the invocation args, thanks!

File: functions/pdf.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *    }
+ *  },
+ * });
+ *
+ * @param args - object - An object with a puppeteer page object, and context.
+ * @param args.page - object - Puppeteer's page object (from await browser.newPage)
+ * @param args.context - object - An object of parameters that the function is called with. See src/schemas.ts
+ */
+module.exports = async function content ({ page, context }) {
+  const { url, html, options } = context;
```
Comment (joelgriffith) [inline functions/pdf.js]: Probably need to drop the console.logs below

Comment (dantheman999301) [reply functions/pdf.js]: Woops! Didn't mean to leave those in.

**Pull request merged**